TCAT/Controllers/RouteDetail+DrawerViewController.swift

```diff
@@ -252,87 +210,166 @@ UIGestureRecognizerDelegate, PulleyDrawerViewControllerDelegate {
+            return indexPath.row > origRow ? result + expandedCount : result
+        }
 
-    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        let newIndexPath = IndexPath(row: indexPath.row + expandedCellCount, section: indexPath.section)
 
-        let direction = directions[indexPath.row]
+        guard let cell = tableView.cellForRow(at: newIndexPath) as? LargeDetailTableViewCell else { return }
 
-        if direction.type == .depart || direction.type == .transfer {
-            let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Cells.largeDetailCellIdentifier) as? LargeDetailTableViewCell
-            cell?.setCell(direction, firstStep: indexPath.row == 0)
-            return cell?.height() ?? RouteDetailCellSize.largeHeight
+        let direction = directions[newIndexPath.row]
+
+        // Flip arrow
+        cell.chevron.layer.removeAllAnimations()
+        cell.isExpanded.toggle()
+
+        let transitionOptionsOne: UIView.AnimationOptions = [.transitionFlipFromTop, .showHideTransitionViews]
+        UIView.transition(with: cell.chevron, duration: chevronFlipDurationTime, options: transitionOptionsOne, animations: {
+            cell.chevron.isHidden = true
+        })
+
+        cell.chevron.transform = cell.chevron.transform.rotated(by: CGFloat.pi)
+        let transitionOptionsTwo: UIView.AnimationOptions = [.transitionFlipFromBottom, .showHideTransitionViews]
+        UIView.transition(with: cell.chevron, duration: chevronFlipDurationTime, options: transitionOptionsTwo, animations: {
+            cell.chevron.isHidden = false
+        })
+
+        // Prepare bus stop data to be inserted / deleted into Directions array
+        var busStops = direction.stops.map { stop in return Direction(name: stop.name, path: direction.path) }
+        let busStopRange = (newIndexPath.row + 1)..<(newIndexPath.row + 1) + busStops.count
+        var indexPathArray = busStopRange.map { i in return IndexPath(row: i, section: 0) }
+
+        tableView.beginUpdates()
+
+        // Insert or remove bus stop data based on selection
+
+        if cell.isExpanded {
+            directions.insert(contentsOf: busStops, at: newIndexPath.row + 1)
+            tableView.insertRows(at: indexPathArray, with: .middle)
+            expandedCellDict[indexPath.row] = indexPathArray.count
         } else {
-            return RouteDetailCellSize.smallHeight
+            directions.removeSubrange(busStopRange)
+            tableView.deleteRows(at: indexPathArray, with: .middle)
+            expandedCellDict.removeValue(forKey: indexPath.row)
         }
 
+        tableView.endUpdates()
+
+        busStops = []
+        indexPathArray = []
     }
+}
 
-    func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
+// MARK: Gesture Recognizers and Interaction-Related Functions
+extension RouteDetailDrawerViewController: UIGestureRecognizerDelegate {
```

Comment:
can we make a `RouteDetailDrawerViewController+Extensions.swift` file and move these extensions into there? Pollo does it a lot if you want a reference